modules/library/main/src/main/java/org/geotools/data/util/GeometryTypeConverterFactory.java

```diff
@@ -162,8 +162,16 @@ else if (source instanceof GeometryCollection)
                     Polygon[] polygons;
                     // NC - Empty Geometry Support
                     if (sourceGeometry.isEmpty()) polygons = new Polygon[0];
-                    else if (source instanceof Polygon) polygons = new Polygon[] {(Polygon) source};
-                    else if (source instanceof GeometryCollection)
+                    else if (source instanceof Polygon) {
+                        // copy polygon
+                        Polygon polygon = (Polygon) source;
```

Comment:
I'm guessing the objective of this copy is to eliminate the srid from the polygon definition?
However the converter is in main, will affect all code using converters, not only GML encoding.
No test is failing, so apparently, either nothing is using that information, or the eventual code path is not tested.
It's not wrong, but I would have fled more comfortable seeing this in the GML encoding modules (limiting the scope of this change).

Reply:
I understand your concern, but only the CRS in the user data of an entire geometry is ever used, never of subgeometries. If you take a multipolygon directly from database or WKT, the CRS is also only set at the root level, so you cannot rely on it being anywhere else either way. (Otherwise we'd see srsname tags pop-up everywhere in subgeometries.) I think it is consistent to always keep the CRS in the root level user data. (Conversely, if we convert from multipolygon to polygon, the converter will set user data in the returned value.)

Changing the encoder would be a fair bit harder because of the way it is implemented with bindings.

Reply:
Since the issue I reported is hypothetical for the moment, let's merge and see how it goes.